tests/e2e/test_templates.py

```diff
@@ -0,0 +1,52 @@
+from pydsl.frontend import template
+from pydsl.type import F32, F64, Bool, Index, SInt8, SInt64, UInt8, UInt64
+
+from pydsl.tensor import Tensor
+from pydsl.arith import min
+
+import numpy as np
+import math
+
+from helper import failed_from
```

Comment:
```suggestion
import math
import numpy as np

from pydsl.arith import min
from pydsl.frontend import template
from pydsl.tensor import Tensor
from pydsl.type import F32, F64, Index, SInt8, UInt64

from helper import failed_from
```
Sort and remove unused.